### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'promotePrefixes'

### ISSUE THREAD

**Issue by Vvegetables** [cackharot/suds-py3#37]
When I send a soap11 request, I throw the above error. Here is my request code:
```
#encoding:utf-8

from suds.client import Client

url = "http://webservice.cingta.com/Calculator/soap11/description/"
headers = {
    "Content-Type": 'application/soap+xml; charset="UTF-8"'
}
client2 = Client(url, headers=headers, faults=False, timeout=15)

result = client2.service.extract_remote_addr()
print(result)
```

**Comment by cackharot** [comment]
Could you share the WSDL to help with the fix?

**Comment by yusufar** [comment]
@cackharot I am getting the same error
The wsdl link is https://rblbank.rupeepower.com/connector/RPCreditCardConnector.wsdl?wsdl
I can provide the code if necessary for debugging.

**Comment by cackharot** [comment]
Try running with DEBUG log, as this might tell what the server response is. 
```
    import logging
    logging.basicConfig(level=logging.INFO)
    logging.getLogger('suds.client').setLevel(logging.DEBUG)
    logging.getLogger('suds.transport').setLevel(logging.DEBUG)
    # logging.getLogger('suds.xsd.schema').setLevel(logging.DEBUG)
```

When I tried the server always returns the WSDL again when we do a SOAP post, so probably some need some server logs too diagnose.

**Comment by yusufar** [comment]
@cackharot Yes i have verified that. but it is working fine with zeep client, so that's why i thought there might be some library issue here.

**Comment by sbywater** [comment]
I am getting this error too, hitting this link: https://ws.paymentech.net/PaymentechGateway/wsdl/PaymentechGateway.wsdl
Note my code works fine under python 2.7, and throws this error when running under 3.6. Hopefully that helps diagnosing.

**Comment by sbywater** [comment]
In my case replyroot.getChildren() returns [Element (prefix=None, name=definitions)] so after setting soapenv = replyroot.getChild('Envelope'), soapenv is None.

**Comment by sbywater** [comment]
For me I was making the request from a wsdl endpoint, so it had no envelope. So I needed to initialize the client with a wsdl endpoint and then change the location before making any calls. I'll add code here in case it helps anyone else:

```
from suds.client import Client

client = Client('https://ws.paymentech.net/PaymentechGateway/wsdl/PaymentechGateway.wsdl`)
client.options.location = 'https://ws.paymentech.net/PaymentechGateway/'
```

**Comment by karimabdelhakim** [comment]
fixed in this [pull request](https://github.com/cackharot/suds-py3/pull/40)

**Comment by tbsf** [comment]
@karamabdelhakim's PR worked for me. @cackharot Any plans to merge this in?

**Comment by cackharot** [comment]
Released a new pypi package with this fix.
https://pypi.org/manage/project/suds-py3/release/1.3.4.0/

**Comment by cackharot** [comment]
This introduced a new bug which @hatchjdecho fixed in #46 . Thanks all who contributed.
